src/GraphQL/Language/CoreToVanillaConverter.cs

```diff
@@ -297,6 +305,14 @@ public NameNode Name(GraphQLName name)
             return new NameNode(name.Value).WithLocation(name, _body);
         }
 
+        public CommentNode Comment(GraphQLComment comment)
+        {
+            if (comment == null)
+                throw new ArgumentNullException(nameof(comment));
+
+            return new CommentNode(comment.Text).WithLocation(comment, _body);
+        }
```

Comment:
```suggestion
        private CommentNode Comment(GraphQLComment comment)
        {
            if (comment == null)
                return null;

            return new CommentNode(comment.Text).WithLocation(comment, _body);
        }
```

And no need in
```c#
frag.CommentNode = source.Comment != null ? Comment(source.Comment) : null;
```
->
```c#
frag.CommentNode = Comment(source.Comment);
```

Reply:
Ok